### PR TITLE
Add WithPath option to set the probe path.

### PR DESCRIPTION
**Pull request by tcnghia**

### CODE REVIEW

File: network/prober/prober_test.go [hunk 1]
```diff
@@ -264,6 +264,54 @@ func TestAsyncMultiple(t *testing.T) {
 	}
 }
 
+func TestWithPathOption(t *testing.T) {
+	path := "/correct/probe/path/"
```
Comment (vagababov) [inline network/prober/prober_test.go]:
```suggestion
const	path = "/correct/probe/path/"
```

File: network/prober/prober_test.go [hunk 2]
```diff
@@ -264,6 +264,54 @@ func TestAsyncMultiple(t *testing.T) {
...
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		t.Logf("Want: %s, Got: %s\n", path, r.URL.Path)
```
Comment (vagababov) [inline network/prober/prober_test.go]: got before want? It's logging, so I'm less nitpicky here :)

File: network/prober/prober_test.go [hunk 3]
```diff
@@ -264,6 +264,54 @@ func TestAsyncMultiple(t *testing.T) {
...
+	defer ts.Close()
+
+	tests := []struct {
+		name    string
+		options []interface{}
+		success bool
+		expErr  bool
+	}{{
+		name:    "no path",
+		options: []interface{}{ExpectsStatusCodes([]int{http.StatusOK})},
```
Comment (vagababov) [inline network/prober/prober_test.go]: you can expect `404` and then get rid of `success` variable altogether?

File: network/prober/prober_test.go (hunk 1 above)
Comment (tcnghia) [reply network/prober/prober_test.go]: Done

File: network/prober/prober_test.go (hunk 2 above)
Comment (tcnghia) [reply network/prober/prober_test.go]: Done

File: network/prober/prober_test.go (hunk 3 above)
Comment (tcnghia) [reply network/prober/prober_test.go]: That's nice.  Done

File: network/prober/prober_test.go [hunk 4]
```diff
@@ -264,6 +264,39 @@ func TestAsyncMultiple(t *testing.T) {
...
+		options: []interface{}{WithPath(path), ExpectsStatusCodes([]int{http.StatusOK})},
+	}, {
+		name:    "wrong path",
+		options: []interface{}{WithPath("/wrong/path"), ExpectsStatusCodes([]int{http.StatusNotFound})},
+	}}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			if ok, _ := Do(context.Background(), network.AutoTransport, ts.URL, test.options...); !ok {
+				t.Errorf("unexpected probe result: %v", ok)
```
Comment (vagababov) [inline network/prober/prober_test.go]: I'd rewrite as 
```suggestion
				t.Error("Unexpected probe failure")
```
but up to you

Comment (tcnghia) [reply network/prober/prober_test.go]: Done

**Comment by tcnghia** [comment]
/hold cancel
@vagababov restamp please

**Pull request merged**